sphinx/builders/linkcheck.py

```diff
@@ -75,18 +74,18 @@ def handle_starttag(self, tag, attrs):
                 self.found = True
 
 
-def check_anchor(f, anchor):
-    """Reads HTML data from a filelike object 'f' searching for *anchor*.
+def check_anchor(response, anchor):
+    """Reads HTML data from a response object 'response' searching for *anchor*.
     Returns True if anchor was found, False otherwise.
     """
     parser = AnchorCheckParser(anchor)
     try:
         # Read file in chunks of 8192 bytes. If we find a matching anchor, we
         # break the loop early in hopes not to have to download the whole thing.
-        chunk = f.read(8192)
-        while chunk and not parser.found:
+        for chunk in response.iter_content(8192):
```

Comment:
The default for `response.__iter__` is much smaller;  any reason we're not using their suggestion?